client/common/services/regex-escape.service.js

```diff
@@ -0,0 +1,19 @@
+(function() {
+    "use strict";
+
+    angular
+        .module("regex-escape", [])
+        .factory("regexEscapeService", regexEscapeService);
+    
+    function regexEscapeService() {
+        var escapeRegex = function(string) {
+            if(string) {
+                return string.replace(/([.*+?^=!:${}()|\[\]\/\\])/g, "\\$1");
+            }
+        };
+        
+        return {
+            escapeRegex: escapeRegex
+        };
+    }
+})();
```

Comment:
I moved the logic for escaping special characters into its own service to
- clean up the search controller
- increase reusability